src/OpenTelemetry/Metrics/Metric.cs

```diff
@@ -124,9 +124,9 @@ public sealed class Metric
 
         internal bool InstrumentDisposed { get; set; }
 
-        public BatchMetricPoint GetMetricPoints()
+        public MetricPointsAccessor GetMetricPointsAccessor()
```

Comment:
unsure if the methodname can still be `GetMetricPoints()` as before...

Reply:
IMHO `MetricPointsAccessor GetMetricPoints()` is better.

Reply:
keeping GetMetricPoints() now.
Its unlikely that anyone would see the name, as exporters typically just do a foreach (var metricpoint in metric.GetMetricPoints())